UnhandledRejection with `metals.super-method-hierarchy` and `metals.goto-super-method` commands

**Bug description**

Hello!

Got problem with coc commands `metals.super-method-hierarchy` and `metals.goto-super-method`
using nvim coc-metals plugin in Scala project.

```
[coc.nvim]: UnhandledRejection: Internal error.
Error: Internal error.
    at Ol (~/.vim/plugged/coc.nvim/build/index.js:37:224)
    at Sp (~/.vim/plugged/coc.nvim/build/index.js:36:11261)
    at Immediate.<anonymous> (~/.vim/plugged/coc.nvim/build/index.js:36:11111)
    at processImmediate (internal/timers.js:464:21)
```

**To Reproduce** Steps:

1. Open `.scala` source code file in neovim
2. Move to parent method of Class / trait (in Scala)
3. Type `:CocCommand` and choose `metals.super-method-hierarchy` or `metals.goto-super-method`
4. Type `:CocOpenLog` to watch errors

**Expected behavior**

Methods of children classes are shown for `metals.super-method-hierarchy`command.
Going to super method for `metals.goto-super-method` command.

**Diagnostics information**

```
:CocOpenLog

2021-11-11T15:42:51.980 INFO (pid:640022) [services] - registered service "languageserver.ccls"
2021-11-11T15:42:51.985 INFO (pid:640022) [coc-conjure] - CoC conjure enabled!
2021-11-11T15:42:52.062 INFO (pid:640022) [coc-git] - Looking for git in: git
2021-11-11T15:42:52.081 INFO (pid:640022) [services] - registered service "highlight"
2021-11-11T15:42:52.605 INFO (pid:640022) [plugin] - coc.nvim 0.0.80-03c9add7cd initialized with node: v14.18.1 after 737ms
2021-11-11T15:42:52.617 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:42:55.138 INFO (pid:640022) [language-client-index] - Language server "metals" started with 640105
2021-11-11T15:42:55.406 INFO (pid:640022) [attach] - receive notification: runCommand []
2021-11-11T15:43:11.982 INFO (pid:640022) [attach] - receive notification: runCommand []
2021-11-11T15:45:05.691 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:06.992 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:07.582 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:08.813 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:11.618 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:13.875 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:15.179 INFO (pid:640022) [attach] - receive notification: highlight []
2021-11-11T15:45:24.608 INFO (pid:640022) [attach] - receive notification: runCommand []
2021-11-11T15:45:27.458 ERROR (pid:640022) [server] - unhandledRejection  Promise {
  │ <rejected> Mm [Error]: Internal error.
  │ │ │ at Ol (~/.vim/plugged/coc.nvim/build/index.js:37:224)
  │ │ │ at Sp (~/.vim/plugged/coc.nvim/build/index.js:36:11261)
  │ │ │ at Immediate.<anonymous> (~/.vim/plugged/coc.nvim/build/index.js:36:11111)
  │ │ │ at processImmediate (internal/timers.js:464:21) {
  │ │ code: -32603,
  │ │ data: 'java.lang.RuntimeException: java.lang.reflect.InvocationTargetException\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)\n' +
  │ │ │ '\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\n' +
  │ │ │ '\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n' +
  │ │ │ '\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\n' +
  │ │ │ '\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\n' +
  │ │ │ '\tat java.base/java.lang.Thread.run(Thread.java:831)\n' +
  │ │ │ 'Caused by: java.lang.reflect.InvocationTargetException\n' +
  │ │ │ '\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n' +
  │ │ │ '\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)\n' +
  │ │ │ '\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n' +
  │ │ │ '\tat java.base/java.lang.reflect.Method.invoke(Method.java:567)\n' +
  │ │ │ '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)\n' +
  │ │ │ '\t... 11 more\n' +
  │ │ │ 'Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4j.TextDocumentIdentifier.getUri()" because the return value of "org.eclipse.lsp4j.                           TextDocumentPositionParams.getTextDocument()" is null\n' +
  │ │ │ '\tat scala.meta.internal.implementation.Supermethods.jumpToSelectedSuperMethod(Supermethods.scala:51)\n' +
  │ │ │ '\tat scala.meta.internal.metals.MetalsLanguageServer.executeCommand(MetalsLanguageServer.scala:1770)\n' +
  │ │ │ '\t... 16 more\n'
  │ }                                                                                                                                                                                           } Mm [Error]: Internal error.
 │ at Ol (~/.vim/plugged/coc.nvim/build/index.js:37:224)
  │ │ at Sp (~/.vim/plugged/coc.nvim/build/index.js:36:11261)
  │ │ at Immediate.<anonymous> (~/.vim/plugged/coc.nvim/build/index.js:36:11111)
  │ │ at processImmediate (internal/timers.js:464:21) {
  │ code: -32603,
  │ data: 'java.lang.RuntimeException: java.lang.reflect.InvocationTargetException\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)\n' +
  │ │ '\tat java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\n' +
  │ │ '\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n' +
  │ │ '\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)\n' +
  │ │ '\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)\n' +
  │ │ '\tat java.base/java.lang.Thread.run(Thread.java:831)\n' +
  │ │ 'Caused by: java.lang.reflect.InvocationTargetException\n' +
  │ │ '\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n' +
  │ │ '\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)\n' +
  │ │ '\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\n' +
  │ │ '\tat java.base/java.lang.reflect.Method.invoke(Method.java:567)\n' +
  │ │ '\tat org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)\n' +
  │ │ '\t... 11 more\n' +
  │ │ 'Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4j.TextDocumentIdentifier.getUri()" because the return value of "org.eclipse.lsp4j.TextDocumentPositionParams.  getTextDocument()" is null\n' +
  │ │ '\tat scala.meta.internal.implementation.Supermethods.jumpToSelectedSuperMethod(Supermethods.scala:51)\n' +
  │ │ '\tat scala.meta.internal.metals.MetalsLanguageServer.executeCommand(MetalsLanguageServer.scala:1770)\n' +
  │ │ '\t... 16 more\n'
  }
```

```
> nvim -v

NVIM v0.5.0
Build type: Debug
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -DNVIM_TS_HAS_SET_MATCH_LIMIT -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=1 -I~/go/src/neovim/build/config -I~/go/src/neovim/src -I~/go/src/neovim/.deps/usr/include -I/usr/include -I~/go/src/neovim/build/src/nvim/auto -I~/go/src/neovim/build/include
Compiled by root@pc

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"
```

```
:CocInfo

## versions
  vim version: NVIM v0.5.0
  node version: v14.18.1
  coc.nvim version: 0.0.80-03c9add7cd
  coc.nvim directory: ~/.vim/plugged/coc.nvim
  term: screen
  platform: linux
```

```
▶ less ~/.vim/plugged/coc.nvim/package.json | grep version
  "version": "0.0.80"
```

```
:CocExtensions

...
* coc-metals 1.0.11 ~/.config/coc/extensions/node_modules/coc-metals
...
```

codeLens is enabled in coc settings:

```
:CocConfig


```json
{
"metals.superMethodLensesEnabled": true,
"metals.enable": true,
"codeLens.enable": true,
"java.implementationsCodeLens.enabled": true,
"diagnostic.virtualText": true,
"java.referencesCodeLens.enabled": true,
"coc.preferences.jumpCommand": "tab drop",
"jedi.enable": true,
"jedi.startupMessage": true,
"jedi.markupKindPreferred": "plaintext",
"jedi.trace.server": true,
"jedi.jediSettings.autoImportModules": [],
"jedi.executable.command": "jedi-language-server",
"jedi.executable.args": [],
"jedi.completion.disableSnippets": false,
"jedi.diagnostics.enable": true,
"jedi.diagnostics.didOpen": true,
"jedi.diagnostics.didChange": true,
"jedi.diagnostics.didSave": true,
"jedi.workspace.extraPaths": [],
"python.pythonPath": "/usr/bin/python3",
"coc.source.iced.enable": true,
"coc.preferences.formatOnType": true,
"coc.preferences.formatOnSaveFiletypes": ["css", "markdown", "javascript", "python", "json", "yaml", "yml"],
"languageserver": {
"ccls": {
"command": "ccls",
"filetypes": [
"c",
"cpp",
"objc",
"objcpp"
],
"rootPatterns": [
".ccls",
"compile_commands.json",
".vim/",
".git/",
".hg/"
],
"initializationOptions": {
"cache": {
"directory": "/tmp/ccls"
}
}
}
}
}
```

`lsp4j` which is used by metals:
```
> ls ~/./.cache/coursier/v1/https/repo1.maven.org/maven2/org/eclipse/lsp4j/org.eclipse.lsp4j/
0.12.0
```

```
> cat /etc/os-release

NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
...
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Tried**

- Updating coc-metals, coc.nvim plugins
- Cleaning .bloop/, .metals/ folders
- Installing the same plugins on Mac Os X.

The problem is still the same.

Have no idea what's the reason of the problem.

--- a/src/index.ts
+++ b/src/index.ts
@@ -365,7 +365,9 @@ async function launchMetals(
         command: ServerCommands.GotoSuperMethod,
         arguments: [
           {
-            document: currentDoc.uri,
+            textDocument: {
+              uri: currentDoc.uri,
+            },
             position,
           },
         ],
@@ -382,7 +384,9 @@ async function launchMetals(
           command: ServerCommands.SuperMethodHierarchy,
           arguments: [
             {
-              document: currentDoc.uri,
+              textDocument: {
+                uri: currentDoc.uri,
+              },
               position,
             },
           ],

